dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/jaxrspropagation/transports/W3CTraceContextPropagatorTestServlet.java

```diff
@@ -0,0 +1,123 @@
+        Map<String, String> carrier = new LinkedHashMap<>();
+        Context context = withSpanContext(
+                                          SpanContext.create(
+                                                             TRACE_ID_BASE16, SPAN_ID_BASE16, TraceFlags.getSampled(), TraceState.getDefault()),
+                                          Context.current());
+        w3cTraceContextPropagator.inject(context, carrier, setter);
+        System.out.println("context " + context);
+        System.out.println("carrier " + carrier);
+        assertThat(carrier.keySet(), contains(TRACE_PARENT));
+        assertThat(carrier.get(TRACE_PARENT), containsString(TRACEPARENT_HEADER_SAMPLED));
+
+    }
+
+    @Test
+    void inject_invalidContext() {
+        Map<String, String> carrier = new LinkedHashMap<>();
+        w3cTraceContextPropagator.inject(
+                                         withSpanContext(
+                                                         SpanContext.create(
+                                                                            TraceId.getInvalid(),
+                                                                            SpanId.getInvalid(),
+                                                                            TraceFlags.getSampled(),
+                                                                            TraceState.builder().put("foo", "bar").build()),
+                                                         Context.current()),
+                                         carrier,
+                                         setter);
+        assertEquals(carrier.size(), 0);
+    }
+
+    private static Context withSpanContext(SpanContext spanContext, Context context) {
+        return context.with(Span.wrap(spanContext));
+    }
+
+    @Test
+    void extract_SampledContext() {
+        Map<String, String> carrier = new LinkedHashMap<>();
+        carrier.put(TRACE_PARENT, TRACEPARENT_HEADER_SAMPLED);
+        assertThat(
+                   getSpanContext(w3cTraceContextPropagator.extract(Context.current(), carrier, getter)), equalTo(
+                                                                                                                  SpanContext.createFromRemoteParent(
+                                                                                                                                                     TRACE_ID_BASE16,
+                                                                                                                                                     SPAN_ID_BASE16,
+                                                                                                                                                     TraceFlags.getSampled(),
+                                                                                                                                                     TraceState.getDefault())));
```

Comment:
Not important right now, but I'd consider adjusting the formatter settings so that it does something more sensible with this.